File: top-interview-150/05-Hashmap/06-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(len(nums)):
            num2 = target - nums[i] 
            if num2 in map.keys():
                return [i, map[num2]]
            else:
                map[nums[i]] = i
        return []

File: top-interview-150/05-Hashmap/06-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numMap = new Dictionary<int, int>();
        int num1, num2;
        for(int i = 0; i < nums.Length; i++)
        {
            num1 = nums[i];
            num2 = target -  num1;
            if(numMap.ContainsKey(num2))
            {
                return [i, numMap[num2]];
            }
            else
            {
                numMap[num1] = i;
            }
        }
        return [];
    }
}